pkg/bloom/bloom.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package bloom
+
+import (
+	"bytes"
+
+	"github.com/bits-and-blooms/bloom"
+)
+
+const FilterErrorRate = 0.01
+
+type Filter interface {
+	Test(string) bool
+	Bytes() []byte
+	Cap() uint
+}
+
+type filter struct {
+	bloom *bloom.BloomFilter
```

Comment:
Why did you opt to use a bloom filter instead of a cuckoo filter? Cuckoo filters work better with low FPP: https://bdupras.github.io/filter-tutorial/

Also, bloom filters will grow as much as needed: https://github.com/bits-and-blooms/bitset/blob/master/bitset.go#L194. I think it is very important for us to limit the memory used by the filter. This is what the cuckoo filter provides for us since before inserting we can tell whether one more element will fit into a fixed-size filter. If it doesn't fit then the caller can make it larger or just fail the insertion. See: https://pkg.go.dev/github.com/seiflotfy/cuckoofilter#section-readme.

What do you think about switching to cuckoo filters from the beginning?

Reply:
Thanks for the heads up, I had heard of cuckoo filters but was not aware of their properties. It's good to know that there is something that works better than bloom filters. 

I've updated the PR to use them instead.